drivers/usb/usbport/endpoint.c

```diff
@@ -957,6 +1147,13 @@ USBPORT_OpenPipe(IN PDEVICE_OBJECT FdoDevice,
             }
         }
 
+        if (Endpoint->TtExtension)
+        {
+            KeAcquireSpinLock(&FdoExtension->TtSpinLock, &OldIrql);
+            RemoveEntryList(&Endpoint->TtLink);
+            KeReleaseSpinLock(&FdoExtension->TtSpinLock, OldIrql);
```

Comment:
If such a similar removal is done in many other places, one can define an "ExInterlockedRemoveEntryList" macro and use it similarly as in:
https://git.reactos.org/?p=reactos.git&a=search&h=HEAD&st=grep&s=ExInterlockedRemoveEntryList